save-plugins/warn-plugin/src/commonMain/kotlin/org/cqfn/save/plugin/warn/utils/Warning.kt

```diff
@@ -103,33 +87,85 @@ internal fun String.extractWarning(warningRegex: Regex,
  * @throws ResourceFormatException when parsing a file
  */
 @Suppress(
     "TooGenericExceptionCaught",
     "SwallowedException",
     "NestedBlockDepth",
+    "LongParameterList",
+    "ReturnCount",
+    "TOO_MANY_PARAMETERS",
 )
 internal fun String.getLineNumber(warningRegex: Regex,
                                   lineGroupIdx: Int?,
                                   placeholder: String,
                                   lineNum: Int?,
+                                  file: Path?,
+                                  linesFile: List<String>?,
 ): Int? {
     val groups = warningRegex.find(this)?.groups ?: return null
 
-    return lineGroupIdx?.let {
-        groups[lineGroupIdx]!!.value.toIntOrNull() ?: run {
-            val lineGroup = groups[lineGroupIdx]!!.value
-            if (lineGroup[0] != placeholder[0]) {
-                throw ResourceFormatException("The group <$lineGroup> is neither a number nor a placeholder.")
-            }
-            try {
-                val line = lineGroup.substringAfterLast(placeholder)
-                lineNum!! + 1 + if (line.isNotEmpty()) line.toInt() else 0
-            } catch (e: Exception) {
-                throw ResourceFormatException("Could not extract line number from line [$this], cause: ${e.describe()}")
+    groups.let {
+        return lineGroupIdx?.let {
+            val lineValue = groups[lineGroupIdx]!!.value
+            if (lineValue.isEmpty()) {
+                return plusLine(file, warningRegex, linesFile, lineNum)
+            } else {
+                lineValue.toIntOrNull() ?: run {
+                    val lineGroup = groups[lineGroupIdx]!!.value
+                    if (lineGroup[0] != placeholder[0]) {
+                        throw ResourceFormatException("The group <$lineGroup> is neither a number nor a placeholder.")
+                    }
+                    try {
+                        val line = lineGroup.substringAfterLast(placeholder)
+                        lineNum!! + 1 + if (line.isNotEmpty()) line.toInt() else 0
+                    } catch (e: Exception) {
+                        throw ResourceFormatException("Could not extract line number from line [$this], cause: ${e.describe()}")
+                    }
+                }
             }
         }
     }
 }
+
+@Suppress(
+    "WRONG_NEWLINES",
+    "TooGenericExceptionCaught",
+    "SwallowedException",
+)
+private fun getRegexGroupSafe(idx: Int?,
+                              groups: MatchGroupCollection,
+                              line: String,
+                              exceptionMessage: String,
+): String? {
+    return idx?.let {
+        try {
+            return groups[idx]!!.value
+        } catch (e: Exception) {
+            throw ResourceFormatException("Could not extract $exceptionMessage from line [$line], cause: ${e.message}")
+        }
+    }
+}
+
+private fun plusLine(
+    file: Path?,
+    warningRegex: Regex,
+    linesFile: List<String>?,
```

Comment:
If we do not have a placeholder, then linesFile can be null, but if we have a placeholder and we got into its logic, then linesFile cannot be null.

Reply:
There is literally no way this method executes normally if linesFile == null. If you perform nullability check elsewhere, then it can be removed from method signature

Reply:
Ok. Fixed